src/main/java/reactor/pool/Pool.java

```diff
@@ -15,21 +15,40 @@
  */
 package reactor.pool;
 
+import java.time.Duration;
+import java.util.function.Function;
+
 import org.reactivestreams.Publisher;
+
 import reactor.core.Disposable;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
-import java.time.Duration;
-import java.util.function.Consumer;
-import java.util.function.Function;
-
 /**
  * A reactive pool of objects.
  * @author Simon Baslé
  */
 public interface Pool<POOLABLE> extends Disposable {
 
+    /**
+     * Warms up the {@link Pool}, if needed. This typically instructs the pool to check for a minimum size and allocate
+     * necessary objects when the minimum is not reached. The resulting {@link Mono} emits the number of extra resources
+     * it created as a result of the {@link PoolBuilder#sizeBetween(int, int) allocation minimum}.
+     * <p>
+     * Note that no work nor allocation is performed until the {@link Mono} is subscribed to.
+     * <p>
+     * Implementations MAY include more behavior, but there is no restriction on the way this method is called by users
+     * (it should be possible to call it at any time, as many times as needed or not at all).
+     *
+     * @apiNote this API is intended to easily reach the minimum allocated size (see {@link PoolBuilder#sizeMin(int)})
+     * without paying that cost on the first {@link #acquire()}. However, implementors should also consider creating
+     * the extra resources needed to honor that minimum during the acquire, as one cannot rely on the user calling
+     * warmup() consistently.
+     *
+     * @return a cold {@link Mono} that triggers resource warmup and emits the number of warmed up resources
+     */
+    Mono<Integer> warmup();
```

Comment:
WDYT about having more generic method like `init` / `start`?
Warmup is too specific action and you may have to add another one in future, for some other initialization, but not warmup

Reply:
on the other hand, I expect users would potentially call this method more than once, eg. have a background thread that regularly ensures the pool is warmed up in terms of minimal amount of allocated objects, instead of only calling this once post construction.